packages/vitest-runner/src/vitest-file-communication.ts

```diff
@@ -0,0 +1,67 @@
+import fs from 'fs/promises';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+import { Mutant } from '@stryker-mutator/api/core';
+
+export type StrykerNamespace = '__stryker__' | '__stryker2__';
+
+export const setupFiles = {
+  coverageFile: resolveSetupFile('__stryker-coverage__.json'),
+  dryRun: resolveSetupFile('dry-run.js'),
+  globalNamespace: resolveSetupFile('global-namespace.js'),
+  hitCountFile: resolveSetupFile('hitCount.txt'),
+  activeMutant: resolveSetupFile('active-mutant.js'),
+  vitestSetup: resolveSetupFile('vitest-setup.js'),
+  hitLimit: resolveSetupFile('hit-limit.js'),
+};
```

Comment:
I realize that this won't work! The `activeMutant` setup file would be shared between all Stryker workers. We should use [`process.env.STRYKER_MUTATOR_WORKER` ](https://stryker-mutator.io/docs/stryker-js/parallel-workers/) to differentiate. Using this for the entire "file-communication protocol" is probably the safest.

Maybe we could change this file to be a class `"FileCommunicator"`. In the constructor, we can let the global environment file be injected. It has methods to read and write from the files. I think they should _not land inside the setup_, instead we should use a temp dir:

```ts
class FileCommunicator {
  private readonly communicationDir = path.resolve(os.tmpdir(), `vitest-runner-${process.env.STRYKER_MUTATOR_WORKER}`);
  constructor(private globalNamespace: string) {
  }

  public async init(){
    await fs.rm(this.communicationDir, { recursive: true, force: true });
  }
  
   //...
}
```

We can simplify further by writing only 1 setup file. This class would be responsible for it

```ts
class FileCommunicator {
 
  get setupFile() {
    return path.resolve(this.communicationDir, 'vitest.setup.js');
  }
  
  get coverageFile() {
     // ....
  }

  async setDryRun(options: DryRunOptions) {
    // ... write content to the setup file
    await fs.writeFile(this.setupFile, `
       // Write hit count, hit limit, isDryRun, global namespace, etc. Altogether in 1 file
    `);
  }
  
   async setMutantRun(options: MutantRunOptions) {
    // ... write content to the setup file
    await fs.writeFile(this.setupFile, `
       // Write active mutant, hit count, hit limit, isDryRun, global namespace, etc. Altogether in 1 file
    `);
  }
```

Reply:
Now that I think of it, using the temp dir probably won't work because you'll need to import from `vitest`.  To that end, only the Current Working Directory (cwd) is valid. We can use `path.resolve` for this purpose.

```ts
class FileCommunicator {
  private readonly communicationDir = path.resolve(`.vitest-runner-${process.env.STRYKER_MUTATOR_WORKER}`);
}
```